aten/src/ATen/native/TensorShape.cpp

```diff
@@ -549,6 +549,58 @@ std::vector<Tensor> broadcast_tensors(TensorList tensors) {
   return expand_outplace(tensors);
 }
 
+static void fastCatOutDim0(const Tensor& out, const MaterializedITensorListRef& inputs) {
+  auto outBytes = out.nbytes();
+  char* dataPtr = reinterpret_cast<char*>(out.data_ptr());
+  size_t totalBytes = 0;
+  for (const Tensor& input : inputs) {
+    if (cat_should_skip_tensor(input)) {
+      continue;
+    }
+    TORCH_CHECK(outBytes >= totalBytes);
```

Comment:
This error check... seems kind of pointless? You are checking that the base of the range you are going to copy into is in bounds... but that's not what you really care about, you care about the end no?